File: backends/fastapi/app/services/avatar_service.py
"""头像目录 —— 直接读 spec/avatars/catalog.json，内存缓存。"""

from __future__ import annotations

import json
from functools import lru_cache

from app.core.config import settings
from app.schemas.avatar import Avatar


@lru_cache(maxsize=1)
def load_avatars() -> list[Avatar]:
    raw = json.loads(settings.avatars_catalog_path.read_text(encoding="utf-8"))
    items = raw.get("avatars", [])
    return [
        Avatar(
            id=a["id"],
            name=a["name"],
            primaryColor=a["primaryColor"],
            svgUrl=f"/static/avatars/{a['id']}.svg",
        )
        for a in items
    ]


def allowed_avatar_ids() -> set[str]:
    return {a.id for a in load_avatars()}

File: backends/nest/src/health/avatar.service.ts
import * as fs from 'fs';
import * as path from 'path';

export interface Avatar {
  id: string;
  name: string;
  primaryColor: string;
  svgUrl: string;
}

let _avatars: Avatar[] = [];
let _loaded = false;

function loadAvatars(): Avatar[] {
  if (_loaded) return _avatars;
  _loaded = true;
  const catalogPath = path.resolve(
    process.env.REPO_ROOT || path.join(__dirname, '../../../../'),
    'spec/avatars/catalog.json',
  );
  try {
    const raw = fs.readFileSync(catalogPath, 'utf-8');
    const data = JSON.parse(raw);
    _avatars = (data.avatars ?? []).map((a: any) => ({
      id: a.id,
      name: a.name,
      primaryColor: a.primaryColor,
      svgUrl: a.svgUrl,
    }));
  } catch {
    _avatars = [];
  }
  return _avatars;
}

export const ALLOWED_AVATAR_IDS: Set<string> = new Set(
  ['neo', 'specter', 'glyph', 'pulse', 'circuit', 'nova', 'oracle', 'drift', 'echo', 'void'],
);

export { loadAvatars };
